### ISSUE TITLE
Move CI to GH actions

### ISSUE THREAD

**Issue by almarklein** [imageio/imageio#589]
Somehow the Travis PR integration is broken (for some time now). We should move to GH actions anyway because Travis crippled the plan for OSS projects. 

I was about to work on this, and then I saw imageio's download stats (2M per month), and my motivation was gone. I have put so much work in this project over the years already. At this point I'm not really a user myself, so why would I put more energy in it now? Perhaps another day I can find the motivation again.

**Comment by FirefoxMetzger** [comment]
🤷 I don't know. For me, 2 million per month sounds like a decent number; if I had a blog with this many page-views, I'd probably just spend all day coding and writing posts xD

Thoughts on increasing popularity: One thing that is on the roadmap is to have `skimage` wrap `imageio`, which could increase usage. Matplotlib's imread reads png natively and forwards the rest to pillow. We could ask if they would be interested in putting `imageio` in between and - in the process - get support for a lot of new formats. Aside from openCV and pillow (which we wrap) that's the 4 main python libraries for reading images (from the top of my head).

**Comment by almarklein** [comment]
I did not mean that I find that number low. I do feel like it's a huge number, even with the knowlegde that most of these are CI builds :)  It's more that it makes me feel some sort of "lonely". There are so many projects/ppl using it, and yet I am the one doing the "dirty work", even though I'm not an actual user myself. Can't blame anyone though.

I don't want to sound like a grumpy old man, but I have many other project that are more appealing for me to work on right now :)  I'm torn between feeling responsible and wondering why it would deserve my precious time.

**Comment by FirefoxMetzger** [comment]
Sounds like Price's law to me; only a hand full of people do the majority of the productive work in any organization. It's the same over at skimage; you see the same names appear again and again. It seems that you are in that minority and provably productive :D

On a more serious note: I understand your frustration; especially if your interests have shifted away from image processing. Maintaining a library isn't the most gratifying work (esp. things like writing CI/CD integrations) and if you're not even getting a personal benefit out of it, it isn't exactly motivating.

From my side, what I can say is that I will be around for at least another year (until I graduate from my PhD program). I'm happy to take on some responsibility during that time. That said, my angle is robotics, ML, and data analysis, not biomedical image processing, so I'm much more interested in things like webcam and video support or thinking about fast integration with common ML frameworks (PyTorch/TensorFlow/Scikit-learn). If that is a future direction for the library you can envision, I'm happy to start discussing the how after we established the new API and integrated it with skimage.

I also don't have many interactions with the more "esoteric" image formats used in bioinformatics. However, our sister labs do (medical image processing) so there might be people willing to contribute there. If you think it is valuable, I can bring it up in our next division meeting.

**Comment by almarklein** [comment]
Thanks! Some help will be appreciated, and your work on the new API means a lot. I hope that in/after that process we can spin some format out into their own repository, so it's easier to maintain the core of imageio, and also easier for others to take responsibility of some of these plugins.

**Comment by FirefoxMetzger** [comment]
Interesting suggestion. I can see the charm in having a smaller code-base resulting, as it, in turn, reduces the time needed for active maintenance of the core repo. It also makes it easier to subscribe to issues that are related to just that one plugin.

Were we to have a really big community that is so big that it becomes hard to coordinate I would totally agree that splitting off plugins may be an option. For us, it seems that we have the opposite problem, so I would vote for keeping things together; this will also bring people together into one place. 

Scattering also comes at the cost of maintaining compatibility between projects. The core has to pull in all the plugins and test, but bugs need to be fixed upstream somewhere; this makes it harder to maintain. If it doesn't how will people get the "full" `imageio` installation?

I would have opted for a path that focuses on adding more documentation and better PR. I did a quick google of imageio the other day, and the results were dysmal: the pipy link is rated more important than our main website. I also googled for "how to open images in python", and the results were skimage, OpenCV, pillow, and matplotlib. Except for pillow, non of these libraries have an I/O focus and `imageio` has everything pillow has (we wrap it after all), but more. There is some great potential buried in the codebase. e.g., webcam access (via `ffmpeg`) without introducing the big OpenCV dependency (which orders color BGR anyway - wtf), video reading (particularly with the new `imiter`, etc. It's just that people don't know about it yet, so we should change that :)

Looking at the topic and the title of the issue, I must say they don't match. Should we change the title to something like "roadmap for imageio" and open a new issue for the GH actions?
